### PR TITLE
Report timeout fix

### LINKED ISSUE #59: To short timeout for generateReport method.
[//]: # (
. Note: for support questions, please use Stackoverflow or Gitter**.
. This repository's issues are reserved for feature requests and bug reports.
.
. In case of any problems with Allure Jenkins plugin** please use the following repository . to create an issue: https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_jenkinsci_allure-2Dplugin_issues&d=DwID-g&c=FXJfUb1oWgygD0uNz-ujnA&r=0UmIrFabaQzxDSWUHSk4XGdSmWxwvG6dNIPErIqoA5w&m=Zk5f8fr-juxF9Wf6APpHK9wfIi4U6IWjAGkBMDwNxsY&s=-jDADsxOnLlZN00vHyYmwMmPaF6fVD-7aZdoIkbNPsU&e=
.
. Make sure you have a clear name for your issue. The name should start with a capital . letter and no dot is required in the end of the sentence. An example of good issue names:
.
. - The report is broken in IE11
. - Add an ability to disable default plugins . - Support emoji in test descriptions
)

#### I'm submitting a ...
  * [x] bug report
  * [ ] feature request
  * [ ] support request => Please do not submit support request here, see note at the top of this template.

#### What is the current behavior?
For large amount of test results, allure framework generates report longer than 60 sec which cause timeout of ExecuteWatchdog. Allure framework process is killed and report is only partially generated. Exception is thrown:

[ERROR] Can't generate allure report data
org.apache.commons.exec.ExecuteException: Process exited with an error: 143 (Exit value: 143)

#### If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem. 
To reproduce the bug you have to generate large amount of data for allure framework. For example: web application tests which stores each send and received messages.

#### What is the expected behavior?
Timeout for AllureCommandline::generateReport should be configurable just like in case of AllureCommandline::serve.

#### What is the motivation / use case for changing the behavior?
Fix the malfunction of the main functionality.

#### Please tell us about your environment:

- Allure version:        2.7.0
- Test framework:        testng@6.14.3
- Allure adaptor:        allure-testng@2.6.0
- Generate report using: allure-maven@2.9
- Apache Maven     3.3.9
- maven-surefire-plugin    2.21.0

#### Other information

[//]: # (
. e.g. detailed explanation, stacktraces, related issues, suggestions . how to fix, links for us to have more context, eg. Stackoverflow, Gitter etc
)

### PR DESCRIPTION
**Pull request by baev**
fixes #59
by @Coder1304

### CODE REVIEW

File: src/main/java/io/qameta/allure/maven/AllureGenerateMojo.java [hunk 1]
```diff
@@ -75,22 +75,10 @@
...
-     */
-    @Parameter(property = "allure.serve.host")
-    protected String serveHost;
-
-    /**
-     * Serve port parameter.
-     */
-    @Parameter(property = "allure.serve.port", defaultValue = "0")
-    protected Integer servePort;
+    @Parameter(property = "allure.report.timeout", defaultValue = "60")
```
Comment (korobochka) [inline src/main/java/io/qameta/allure/maven/AllureGenerateMojo.java]:
```suggestion
    @Parameter(property = "allure.report.timeout", defaultValue = "600")
```
Can I suggest a longer default value? 1 minute is not enough for big reports on Windows machines.

Comment (Coder1304) [reply src/main/java/io/qameta/allure/maven/AllureGenerateMojo.java]: I know it's to short, that's why this pull was created. This value will be configurable from now, so you would be able to set it to any value you like. 60sec worked here for a long time and from all allure-maven users just two of us starts to complains about that. @baev which default value would be sufficient in your opinion?

Comment (baev) [reply src/main/java/io/qameta/allure/maven/AllureGenerateMojo.java]: I'm ok with one minute timeout by default

File: src/main/java/io/qameta/allure/maven/AllureServeMojo.java [hunk 2]
```diff
@@ -22,6 +22,24 @@
 @Mojo(name = "serve", defaultPhase = LifecyclePhase.SITE)
 public class AllureServeMojo extends AllureGenerateMojo {
 
+    /**
+     * Serve timeout parameter in seconds.
+     */
+    @Parameter(property = "allure.serve.timeout", defaultValue = "3600")
+    private String serveTimeout;
```
Comment (baev) [inline src/main/java/io/qameta/allure/maven/AllureServeMojo.java]: I don't like this to be a string as well as new introduced report timeout to be a string

Comment (Coder1304) [reply src/main/java/io/qameta/allure/maven/AllureServeMojo.java]: @baev I send the patch on `allure@qameta.io` with `gh-59-fix-proposition` in email subject.

Comment (baev) [reply src/main/java/io/qameta/allure/maven/AllureServeMojo.java]: @Coder1304 still got no email from you, could you please resend it?

Comment (Coder1304) [reply src/main/java/io/qameta/allure/maven/AllureServeMojo.java]: @baev I've send it again to `allure@qameta.io` with `gh-59-fix-proposition` in the subject. And as a reply for the first mail I send you.

**Comment by baev** [comment]
Looks good to me.

In order to merge this:

* @Coder1304 you need to accept our CLA https://cla-assistant.io/allure-framework/allure-maven?pullRequest=60

* Then you need to add email address (ends with `**yk.ctr@sabre.com`) to your Github account ( https://github.com/settings/emails page ) in order to link commits with your Github account

**Comment by Coder1304** [comment]
@baev I realized that I forgot to change private data. Could you change my private data to this:
```
git checkout gh-59-fix-proposition
git reset --soft e8e10de
git commit -m "report timeout fix" --amend --author="Coder1304 <Coder1304.public@gmail.com>"
git push -f origin HEAD:gh-59-fix-proposition
```
I would be more than happy if do this.

**Comment by baev** [comment]
@Coder1304 done, now all we need is CLA

**Comment by eroshenkoam** [comment]
@Coder1304 
1. It will be ready tomorrow)
2. Just create PR in this repo)
3. Fixed: https://github.com/allure-framework/allure-maven/commits/master

**Comment by Coder1304** [comment]
@eroshenkoam Could you change my data to:
`--author="Coder1304 <Coder1304.public@gmail.com>"`

**Comment by eroshenkoam** [comment]
done) ty for pr 👍

**Comment by Coder1304** [comment]
I have to prepare the change first. Thanks.

**Comment by Coder1304** [comment]
@eroshenkoam 
I am getting problem to push anything to you repo.
```
$ git push -v origin HEAD:some_branch_here
Pushing to git@github.com:allure-framework/allure-maven.git
ERROR: Permission to allure-framework/allure-maven.git denied to Coder1304.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
``` 
Tried with ssh and https, same result.

**Comment by baev** [comment]
@Coder1304 see https://help.github.com/articles/creating-a-pull-request/

> If you don't have write access to the repository where you'd like to create a pull request, you must create a fork, or copy, of the repository first. For more information, see ["Creating a pull request from a fork"](https://help.github.com/articles/creating-a-pull-request-from-a-fork) and ["About forks."](https://help.github.com/articles/about-forks)

**Comment by Coder1304** [comment]
@baev Thanks a lot for this link.

**Pull request merged**